subprojects/diagnostics/src/main/java/org/gradle/configuration/Help.java

```diff
@@ -35,7 +36,18 @@
 
 @DisableCachingByDefault(because = "Produces only non-cacheable console output")
 public class Help extends DefaultTask {
-    private String taskPath;
+    private final Property<String> taskPath = getObjectFactory().property(String.class);
+
+    private final Property<TaskDetailsModel> taskModel = getObjectFactory().property(TaskDetailsModel.class).convention(taskPath.map(this::mapFromTaskPath));
```

Comment:
Note that I looked into marking `taskModel` as `finalizeValueOnRead()` in an initialization block, however I am still not clear what the goal of that is in our use case, so I removed that bit. I wondered if it was to avoid having the transformer invoked twice, but in my tests, the transformer was still invoked twice, once during cache storing, and another during execution (maybe that is expected, and the intent is then probably something else).

```
java.lang.Throwable: Mapping showTask
        at org.gradle.configuration.Help.mapFromTaskPath(Help.java:158)
        at org.gradle.api.internal.provider.TransformBackedProvider.mapValue(TransformBackedProvider.java:73)
        at org.gradle.api.internal.provider.TransformBackedProvider.calculateExecutionTimeValue(TransformBackedProvider.java:57)
        at org.gradle.api.internal.provider.MappingProvider.calculateExecutionTimeValue(MappingProvider.java:64)
        at org.gradle.configurationcache.serialization.codecs.FixedValueReplacingProviderCodec.encodeProvider(ProviderCodecs.kt:72)
        at org.gradle.configurationcache.serialization.codecs.PropertyCodec.encode(ProviderCodecs.kt:249)
        at org.gradle.configurationcache.serialization.codecs.PropertyCodec.encode(ProviderCodecs.kt:241)
        at org.gradle.configurationcache.serialization.codecs.BindingsBackedCodec.encode(BindingsBackedCodec.kt:52)
        at org.gradle.configurationcache.serialization.DefaultWriteContext.write(Contexts.kt:85)
        at org.gradle.configurationcache.serialization.beans.BeanPropertyWriterKt.writeNextProperty(BeanPropertyWriter.kt:96)
        at org.gradle.configurationcache.serialization.beans.BeanPropertyWriter.writeStateOf(BeanPropertyWriter.kt:55)
        ...
```

```
java.lang.Throwable: Mapping showTask
        at org.gradle.configuration.Help.mapFromTaskPath(Help.java:158)
        at org.gradle.api.internal.provider.TransformBackedProvider.mapValue(TransformBackedProvider.java:73)
        at org.gradle.api.internal.provider.TransformBackedProvider.calculateOwnValue(TransformBackedProvider.java:65)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:102)
        at org.gradle.api.internal.provider.MappingProvider.calculateOwnValue(MappingProvider.java:55)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:102)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.withFinalValue(AbstractMinimalProvider.java:158)
        at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:133)
        at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:26)
        at org.gradle.api.internal.provider.AbstractProperty.finalizeNow(AbstractProperty.java:240)
        at org.gradle.api.internal.provider.AbstractProperty.beforeRead(AbstractProperty.java:232)
        at org.gradle.api.internal.provider.AbstractProperty.calculateOwnValue(AbstractProperty.java:128)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.get(AbstractMinimalProvider.java:83)
        at org.gradle.configuration.Help.printTaskHelp(Help.java:99)
        at org.gradle.configuration.Help.displayHelp(Help.java:92)
...
```

Reply:
It should still be marked `finalizeValueOnRead` because that's the intent. The underlying bug will eventually be fixed.

Reply:
> It should still be marked `finalizeValueOnRead` because that's the intent. The underlying bug will eventually be fixed.

Happy to oblige, however, while I think I understand the mechanics, I still don't get what the intent is for this specific case. What are we aiming to achieve?